tracer/test/test-applications/integrations/Samples.InstrumentedTests/Vulnerabilities/EmailHtmlInjection/EmailHtmlInjectionTests.cs

```diff
@@ -0,0 +1,286 @@
+    }
+
+    // Tests for method Send(MailMessage message);
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailMessageTaintedVaulesHtml_ThenIsVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, taintedName, taintedLastName);
+        TestEmailSendCall(() => Send(mailMessage));
+        AssertVulnerable(emailHtmlInjectionType, "Hi :+-Alice<h1>Hi</h1>-+: :+-Stevens-+:!");
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailMessageTaintedVaulesHtmlEscaped_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, WebUtility.HtmlEncode(taintedName), WebUtility.HtmlEncode(taintedLastName));
+        TestEmailSendCall(() => Send(mailMessage));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailMessageTaintedVaulesHtmlEscaped_ThenIsNotVulnerable2()
+    {
+        var mailMessage = BuildMailMessage(true, HttpUtility.HtmlEncode(taintedName), HttpUtility.HtmlEncode(taintedLastName));
+        TestEmailSendCall(() => Send(mailMessage));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailMessageNotTaintedVaulesHtml_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, untaintedName, untaintedLastName);
+        TestEmailSendCall(() => Send(mailMessage));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailMessageTaintedVaulesNoHtml_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(false, taintedName, taintedLastName);
+        TestEmailSendCall(() => Send(mailMessage));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailMessageNull_ThenIsNotVulnerable()
+    {
+        try
+        {
+            Send(null);
+        }
+        catch (ArgumentNullException) { }
+        
+        AssertNotVulnerable();
+    }
+
+    private void Send(MailMessage mailMessage)
+    {
+        using (var client = BuildClient())
+        {
+            client.Send(mailMessage);
+        }
+    }
+
+    // Tests for method SendMailAsync(MailMessage message);
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageTaintedVaulesHtml_ThenIsVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, taintedName, taintedLastName);
+        TestEmailSendCall(() => SendMailAsync(mailMessage));
+        AssertVulnerable(emailHtmlInjectionType, "Hi :+-Alice<h1>Hi</h1>-+: :+-Stevens-+:!");
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageTaintedVaulesHtmlEscaped_ThenIsNotVulnerable()
+    {
+        var mailMessage = BuildMailMessage(true, WebUtility.HtmlEncode(taintedName), WebUtility.HtmlEncode(taintedLastName));
+        TestEmailSendCall(() => SendMailAsync(mailMessage));
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnEmail_WhenSendMailAsyncMailMessageTaintedVaulesHtmlEscaped_ThenIsNotVulnerable2()
+    {
```

Comment:
wondering if these tests could be one single theory test?

Reply:
We have not tried theory tests in the instrumented tests, whose IL is modified and work with tainted/untainted data. It will be worth to refactor them or take a look at the idea. I will add it to my list of possible tasks. Thanks!